src/efi-application.c

```diff
@@ -288,10 +290,20 @@ __tpm_event_efi_bsa_rehash(const tpm_event_t *ev, const tpm_parsed_event_t *pars
 		return tpm_event_get_digest(ev, ctx->algo->openssl_name);
 	}
 
+	/* The next boot can have a different kernel */
+	if (sdb_is_kernel(evspec->efi_application)) {
+		new_application = sdb_get_next_kernel();
+		if (new_application) {
+			memcpy(&evspec_clone, evspec, sizeof(evspec_clone));
```

Comment:
I'd use evspec_clone = *evspec; instead. Does the same thing but you get type checking for free.

Reply:
Good idea. Done!